mathics/builtin/box/inout.py

```diff
@@ -465,40 +469,37 @@ def apply_list(self, boxes, evaluation):
         return result
 
     def init(self, *items, **kwargs):
-        # TODO: check that each element is an string or a BoxConstruct
+        # TODO: check that each element is an string or a BoxExpression
         self.box_options = {}
         if isinstance(items[0], Expression):
             if len(items) != 1:
                 raise Exception(
-                    items, "is not a List[] or a list of Strings or BoxConstruct"
+                    items, "is not a List[] or a list of Strings or BoxExpression"
                 )
             if items[0].has_form("List", None):
                 items = items[0]._elements
             else:
                 raise Exception(
-                    items, "is not a List[] or a list of Strings or BoxConstruct"
+                    items, "is not a List[] or a list of Strings or BoxExpression"
                 )
         for item in items:
-            if not isinstance(item, BoxConstruct):
+            if isinstance(item, String):
+                item = _BoxedString(item.value)
+            if not isinstance(item, BoxExpression):
                 raise Exception(
-                    item, "is not a List[] or a list of Strings or BoxConstruct"
+                    item, "is not a List[] or a list of Strings or BoxExpression"
                 )
         self.items = tuple(items)
-
-        elements = []
-        for item in items:
-            if isinstance(item, BoxConstruct):
-                elements.append(item.to_expression())
-            else:
-                elements.append(item)
-        elements = Expression(SymbolList, *elements)
-        self._elements = tuple((elements,))
+        self._elements = None
 
     def to_expression(self):
         """
         returns an evaluable expression.
         """
-        items = (item.to_expression() for item in self.items)
+        items = tuple(
+            item.to_expression() if isinstance(item, BoxExpression) else item
+            for item in self.items
+        )
         result = Expression("RowBox", Expression("List", *items))
```

Comment:
RowBox doesn't work here?

Reply:
This is the `to_expression` method, so what we want here is an `Expression` object. This is needed because during the evaluation, there are   expressions like
```
expr = Expression(Symbol("F"), RowBox(String("bla")))
```
which could appear when the user defines a custom `MakeBoxes` rule. 

Then, maybe `F` has specific rules when acting over a `RowBox`, so, when you call 
```
expr.evaluate(evaluation)
```
you need the interface of an expression. The methods are then implemented by calling `to_expression` and then delegating the method to the `Expression` you get there object.

Reply:
This makes perfect sense. Thanks for the explanation. 

Given this that we need an Expression, then please adjust the `Expression()` calls to the appropriate form. For example use the pre-defined SymbolRowBox that appears in `mathcis.core.systemsymbols`, or turn these into top-level function calls `to_epxression()` calls. For the list expression, if we can a ListExpression here and that works, that would be nice, otherwise as a fallback either use `to_expression()` or `SymbolList()` as before. Thanks.

Reply:
Ok I see this change in https://github.com/Mathics3/mathics-core/pull/313  

So this PR is obsolete if we just merge the other one in, right?

Reply:
@rocky,  Now, this has some small changes regarding the other. Once I finish with the changes that you asked for, I can merge this to master, or wait that you to merge #308 and then merge this.

Reply:
I've just merged #308. So if you could rebase off of master and then finish that would be great. Thanks!

Reply:
OK.